include/zephyr/net/sntp.h

```diff
@@ -86,13 +86,31 @@ void sntp_close(struct sntp_ctx *ctx);
  *
  * @param server Address of server in format addr[:port]
  * @param timeout Query timeout
- * @param time Timestamp including integer and fractional seconds since
+ * @param ts Timestamp including integer and fractional seconds since
  * 1 Jan 1970 (output).
  *
  * @return 0 if ok, <0 if error (-ETIMEDOUT if timeout).
  */
 int sntp_simple(const char *server, uint32_t timeout,
-		struct sntp_time *time);
+		struct sntp_time *ts);
+
+/**
+ * @brief Convenience function to query SNTP in one-shot fashion
+ * using a pre-initialized socket
+ *
+ * Convenience wrapper which calls sntp_init(), sntp_query() and
+ * sntp_close().
+ *
+ * @param addr IP address of NTP/SNTP server.
+ * @param addr_len IP address length of NTP/SNTP server.
+ * @param timeout Query timeout
+ * @param ts Timestamp including integer and fractional seconds since
+ * 1 Jan 1970 (output).
+ *
+ * @return 0 if ok, <0 if error (-ETIMEDOUT if timeout).
+ */
+int sntp_simple_sock(struct sockaddr *addr, socklen_t addr_len, uint32_t timeout,
```

Comment:
The `sock` feels wrong here as it gives impression about socket but as this is for address setting variant, we could call this `sntp_simple_addr()` for example.